src/components/movement/CameraRig.tsx

```
@@ -0,0 +1,65 @@
+ * Camera rig that smoothly follows a target using OrbitControls
+ */
+const CameraRig: React.FC<CameraRigProps> = ({
+  target,
+  followSpeed = 0.5,
+  minDistance = 2,
+  maxDistance = 4,
+  maxPolarAngle = Math.PI / 2, // Prevents camera from going below the horizontal plane
+  children
+}) => {
+  // Create a ref for orbit controls
+  const controlsRef = useRef<any>(null);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)